_rules/element-lang-valid-de46e4.md

```diff
@@ -141,22 +186,52 @@ The rule applies to elements within the `body` of a webpage. `html` elements are
 
 ```html
 <html lang="en">
-	<body></body>
+	<body>
+		They wandered into a strange Tiki bar on the edge of the small beach town.
+	</body>
 </html>
 ```
 
 #### Inapplicable Example 2
 
-An empty value for the `lang` attribute is ignored, as the rule only applies to `lang` attributes that are not empty ("").
+An empty `lang` [attribute value][] is ignored, as the rule only applies to `lang` attributes that are not empty (`""`).
+
+```html
+<html>
+	<body>
+		<article lang="">
+			They wandered into a strange Tiki bar on the edge of the small beach town.
+		</article>
+	</body>
+</html>
+```
+
+#### Inapplicable Example 3
+
+The `lang` [attribute value][] does not have a valid language tag, and does not have a [visible][] [text node][] as a [descendant][] in the [flat tree][].
 
 ```html
 <html>
 	<body>
-		<article lang=""></article>
+		<p lang="hidden">
+			<span style="display: none;">
+				They wandered into a strange Tiki bar on the edge of the small beach town.
+			</span>
+		</p>
 	</body>
 </html>
 ```
 
+[node document]: https://dom.spec.whatwg.org/#concept-node-document
+[content type]: https://dom.spec.whatwg.org/#concept-document-content-type
+[descendant]: https://dom.spec.whatwg.org/#concept-tree-descendant
+[flat tree]: https://drafts.csswg.org/css-scoping/#flat-tree
 [grandfathered tags]: https://tools.ietf.org/html/bcp47#section-2.2.8
 [bcp 47]: https://tools.ietf.org/html/bcp47#section-2.1
 [valid language tag]: #valid-language-tag
+[whitespace]: #whitespace 'Definition of Whitespace'
+[visible]: #visible 'Definition of visible'
+[text node]: https://dom.spec.whatwg.org/#text
+[descendant]: https://dom.spec.whatwg.org/#concept-tree-descendant
+[attribute value]: #attribute-value 'Definition of Attribute Value'
+[included in the accessibility tree]: #included-in-the-accessibility-tree
```

Comment:
```suggestion
[attribute value]: #attribute-value 'Definition of Attribute Value'
[bcp 47]: https://tools.ietf.org/html/bcp47#section-2.1
[content type]: https://dom.spec.whatwg.org/#concept-document-content-type
[descendant]: https://dom.spec.whatwg.org/#concept-tree-descendant
[flat tree]: https://drafts.csswg.org/css-scoping/#flat-tree
[grandfathered tags]: https://tools.ietf.org/html/bcp47#section-2.2.8
[included in the accessibility tree]: #included-in-the-accessibility-tree
[node document]: https://dom.spec.whatwg.org/#concept-node-document
[text node]: https://dom.spec.whatwg.org/#text
[valid language tag]: #valid-language-tag
[visible]: #visible 'Definition of visible'
[whitespace]: #whitespace 'Definition of Whitespace'
```
If it had been ordered in the first place, you'd probably have noticed the duplicate reference to "descendant" and clean it…